### ISSUE TITLE
Too many open files (os error 24) on Linux

### ISSUE THREAD

**Issue by erg** [BurntSushi/ripgrep#650]
I have a modest amount of files (226,441) from ec2 in subdirectories and ripgrep opens too many at once on Arch Linux.

```
rg asdfasdfasdf
...
./ec2-stuff/2016/09/28/03: Too many open files (os error 24)
./ec2-stuff//2016/09/13/07: Too many open files (os error 24)
./ec2-stuff///2016/09/28/13: Too many open files (os error 24)
./ec2-stuff//2016/09/07/18: Too many open files (os error 24)
./ec2-stuff//2016/09/28/07: Too many open files (os error 24)
./ec2-stuff//2016/09/07/15: Too many open files (os error 24)
./ec2-stuff//2016/09/07/06: Too many open files (os error 24)
...
```

```
➜  ec2 find . | wc -l
226441
➜  ~ rg --version
ripgrep 0.7.0
-AVX -SIMD
➜  ~ uname -a
Linux ergbox 4.13.8-1-ARCH #1 SMP PREEMPT Wed Oct 18 11:49:44 CEST 2017 x86_64 GNU/Linux
➜  ~ cat /proc/sys/fs/file-max
13166298
➜  ~ ulimit -Hn               
4096
➜  ~ ulimit -Sn               
1024
```

**Comment by okdana** [comment]
This was fixed in 0.7.1 — see #648

**Comment by erg** [comment]
Ah. Thanks!
